woocommerce/woocommerce-ios
# Promote extensions on Product Filter

Closes: #10435 

# Why

As part of the product extension project, this PR adds an "Explore button" to the product types filter whose correspondent extension is not yet installed on the store.

This button will redirect the user to the extension's webpage with the idea of promoting it.

# How

- Create a new `PromotableProductType` to be used in the product filters instead of the current `ProductType`
- Show an "Explore" button only to the `PromotableProduct` types.
- Makes use of a `ResultsController` to fetch the plugin status synchronously.

I ended up abstracting the `PromotableProductType` behind the existing `FilterType`. I'm not very sure if this is the correct way to extend the product filters. 

They are meant to behave quite generically so adding specificity to the `FilterType` protocols didn't feel good. In the same way, creating a specific view controller only for this change felt like an overkill 🤷 

# Demo


https://github.com/woocommerce/woocommerce-ios/assets/562080/355377e5-633d-470c-82fb-ff254264fc22

# Testing Steps

- Launch the app and navigate to the products tab
- Tap on the filter button then on the Product Type section.
- See that types that are part of an extension that you don't have installed are not selectable and have an explore button.
- Tap the explore button and see that you are redirected to the extension webpage

---
